src/uiSelectChoicesDirective.js

```diff
@@ -68,7 +66,7 @@ uis.directive('uiSelectChoices',
         scope.$watch('$select.search', function(newValue) {
           if(newValue && !$select.open && $select.multiple) $select.activate(false, true);
           $select.activeIndex = $select.tagging.isActivated ? -1 : 0;
-          if (!attrs.minimumInputLength || $select.search.length >= attrs.minimumInputLength) {
+          if ((!attrs.minimumInputLength || $select.search.length >= attrs.minimumInputLength) && $select.open) {
```

Comment:
Okay, if that is the case then we can revert almost everything. Only thing we need to add is when `activating/focus` the control we need to execute the `refresh `function.

The users can already check if it is open by adding (as you already mentioned)  a second param `$select.open`